src/c.c

```diff
@@ -274,36 +273,41 @@ r_obj* vec_c_fallback(r_obj* ptype,
                       struct r_lazy error_call) {
   r_obj* cls = KEEP(r_attrib_get(ptype, syms_fallback_class));
   bool implements_c = class_implements_base_c(cls);
-  FREE(1);
 
   if (implements_c) {
+    FREE(1);
     return vec_c_fallback_invoke(xs, name_spec, error_call);
-  } else {
```

Comment:
No changes here, just using an early return in `vec_c_fallback()` to flatten a level of nesting, better matching the new `list_unchop_c_fallback()`